src/webgl/p5.Framebuffer.js

```diff
@@ -931,6 +950,238 @@ class Framebuffer {
+   * getting an image, the x and y parameters define the coordinates for the
+   * upper-left corner of the image, regardless of the current <a href="#/p5/imageMode">imageMode()</a>.
+   *
+   * @method get
+   * @param  {Number}         x x-coordinate of the pixel
+   * @param  {Number}         y y-coordinate of the pixel
+   * @param  {Number}         w width
+   * @param  {Number}         h height
+   * @return {p5.Image}       the rectangle <a href="#/p5.Image">p5.Image</a>
+   */
+  /**
+   * @method get
+   * @return {p5.Image}      the whole <a href="#/p5.Image">p5.Image</a>
+   */
+  /**
+   * @method get
+   * @param  {Number}        x
+   * @param  {Number}        y
+   * @return {Number[]}      color of pixel at x,y in array format [R, G, B, A]
+   */
+  get(x, y, w, h) {
+    const colorFormat = this._glColorFormat();
+    if (x === undefined && y === undefined) {
+      x = 0;
+      y = 0;
+      w = this.width;
+      h = this.height;
+    } else if (w === undefined && h === undefined) {
+      if (x < 0 || y < 0 || w >= this.width || h >= this.height) {
+        return [0, 0, 0, 0];
+      }
+
+      return readPixelWebGL(
+        this.gl,
+        this.framebuffer,
+        x * this.density,
+        y * this.density,
+        colorFormat.format,
+        colorFormat.type
+      );
+    }
+
+    const rawData = readPixelsWebGL(
+      undefined,
+      this.gl,
+      this.framebuffer,
+      x * this.density,
+      y * this.density,
+      w * this.density,
+      h * this.density,
+      colorFormat.format,
+      colorFormat.type
+    );
+    // Framebuffer data might be either a Uint8Array or Float32Array
+    // depending on its format, and it may or may not have an alpha channel.
+    // To turn it into an image, we have to normalize the data into a
+    // Uint8ClampedArray with alpha.
+    const fullData = new Uint8ClampedArray(
+      w * h * this.density * this.density * 4
+    );
+    const channels = colorFormat.type === this.gl.RGB ? 3 : 4;
+    for (let y = 0; y < h * this.density; y++) {
+      for (let x = 0; x < w * this.density; x++) {
+        for (let channel = 0; channel < 4; channel++) {
+          const idx = (y * w * this.density + x) * 4 + channel;
+          if (channel >= channels) {
+            fullData[idx] = 255;
+          } else {
+            const prevIdx = channels === 4
+              ? idx
+              : (y * w * this.density + x) * channels + channel;
+            fullData[idx] = rawData[prevIdx];
+          }
+        }
+      }
+    }
+
+    // Create an image from the data
+    const region = new p5.Image(w * this.density, h * this.density);
+    region.imageData = region.canvas.getContext('2d').createImageData(
+      region.width,
+      region.height
+    );
+    region.imageData.data.set(fullData);
+    region.pixels = region.imageData.data;
+    region.updatePixels();
+    if (this.density !== 1) {
+      // TODO: support get() at a pixel density > 1
```

Comment:
Out of curiosity, how much extra work would this take?

Reply:
This would mean making p5.Image support pixel density. I'd have to hunt through the p5 source a bit to confirm, but potentially not that much? It already has a `_density` property hardcoded to 1 that would get used in `p5.Image#get`, so it might just be a matter of updating its internal canvas size accordingly. But I suspect there are also some other bits (like the 2D mode tint() implementation) that read width/height with an assumption that the pixel density is 1.

Reply:
Sounds like it could be potentially spidery in solving this. Maybe a good task for the future?

Reply:
Made an issue for it here: https://github.com/processing/p5.js/issues/6114